engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala

```diff
@@ -35,16 +35,29 @@ class LanguageServerComponent(config: LanguageServerConfig)
   /** @inheritdoc */
   override def start(): Future[ComponentStarted.type] = {
     logger.info("Starting Language Server...")
+    val module = new MainModule(config)
+    val initMainModule =
+      for {
+        _ <- module.init
+        _ <- Future { logger.debug("Main module initialized") }
+      } yield ()
+    val bindJsonServer =
+      for {
+        binding <- module.jsonRpcServer.bind(config.interface, config.rpcPort)
+        _       <- Future { logger.debug("Json RPC server initialized") }
+      } yield binding
+    val bindBinaryServer =
+      for {
+        binding <- module.binaryServer.bind(config.interface, config.dataPort)
+        _       <- Future { logger.debug("Binary server initialized") }
+      } yield binding
```

Comment:
The module's initialization and socket binding will run concurrently. I'm not sure if it is the behavior you wanted to implement.

Reply:
They are independent actions, and I indeed wanted to run them in parallel. This way it shaves ~2 seconds off the startup time on my machine.